packages/react-ui-core/src/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,214 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import debounce from 'lodash/debounce'
+import { randomId } from '@rentpath/react-ui-utils'
+import themed from 'react-themed'
+import { Input } from '../Form'
+import List from '../List'
+import ItemWrapper from './ItemWrapper'
+
+const ESC = 27
+const ENTER = 13
+const ARROW_UP = 38
+const ARROW_DOWN = 40
+
+@themed(/^SearchBar/, {
+  pure: true,
+})
+
+export default class SearchBar extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    theme: PropTypes.object,
+    onInput: PropTypes.func,
+    suggestions: PropTypes.node,
+    suggestionsVisible: PropTypes.bool,
+    resetButton: PropTypes.node,
+    onAfterReset: PropTypes.func,
+    submitButton: PropTypes.node,
+    onSubmit: PropTypes.func,
+    value: PropTypes.string,
+    placeholder: PropTypes.string,
+    suggestionValueSelector: PropTypes.func,
+  }
+
+  static defaultProps = {
+    theme: {},
+    value: '',
+    onAfterReset: () => {},
+    onSubmit: () => {},
+    onInput: () => { },
+    suggestionsVisible: false,
+  }
+
+  constructor(props) {
+    super(props)
+
+    const { value, suggestionsVisible } = this.props
+
+    this.reset = this.reset.bind(this)
+    this.handleChange = this.handleChange.bind(this)
```

Comment:
Seems like there is way too much going on with binding. I wonder if we really need all of these?

Reply:
I will double check, but I fairly sure those were added one by one as needed. These are mostly passed down to child components through props and need proper context when invoked.